src/Tasks/ResourceHandling/InputFormatNotSupportedException.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Build.Tasks.ResourceHandling
+{
+    [Serializable]
+    internal class InputFormatNotSupportedException : Exception
```

Comment:
Is there a reason you can't use `FormatException` or `ArgumentException` instead of making your own (empty) new type of exception?

Reply:
Having a custom exception type makes detecting this error condition easy using a catch-specific-type block.

Reply:
Tracing how it's currently caught, I believe a FormatException will still have the exact same effect.

Also, I don't see anywhere that you try to catch it; do you have plans for adding something like that in the future? If not, you're breaking something that could otherwise be caught by an ArgumentException, failing more cleanly.

Reply:
> I believe a FormatException will still have the exact same effect.

Since it's a standard exception type, other code could insert a `FormatException` that would then be inadvertently caught and handled as this specific problem, rather than handled at another layer.

> Also, I don't see anywhere that you try to catch it

https://github.com/microsoft/msbuild/pull/4830/files#diff-1d2aacce8c88cda3235c1659ea67e164R2606-R2611

> If not, you're breaking something that could otherwise be caught by an ArgumentException, failing more cleanly.

This is harder to distinguish since other places could throw `ArgumentException` inside the big `ProcessFile` catch block.